fecgov/fec-eregs
# update node v10.16.0 and python v3.7.5

**Summary:**

`grunt` pkg  was updated on  **Mar 16th 2020** which broke the grunt scripts( compile the front end assets) in `fec-eregs` repo. This grunt pkg ONLY supports NODE >= 8. (see # https://gruntjs.com/blog/2020-03-16-grunt-1.1.0-released).
I upgraded node to LTS v10.16.0 `fec-eregs` repo and also upgraded python v3.7.5.  Deployed `feature/update-pkgs` branch via CircleCI.  Build ran OK and the eregs app deployed  successfully to `dev` space 

**Summary of changes:**

1. Resolves https://github.com/fecgov/fec-eregs/issues/480
2. Update the node LTS  v10.16.0
3. Update python to v3.7.5
4. Update grunt and mkdirp pkgs that were causing the build fails (regulation repo)
5. Deleted  unused and deprecated pkg
6. Removed deprecated warning from showing up on the deploy scripts

**How to test:**

1. Activate new python 3.7.5 virtual env : `pyenv virtualenv 3.7.5 veregs375`
2. Alias node new version to default : `nvm alias default v10.16.0`
3. Use new version: `nvm use default`
4. Upgrade npm: `npm install -g npm`
5. Check node version: `node -v`
6. Check npm version : `npm -v`
7. At this point your node and npm should be : v10.16.0 & v6.14.3
8. Remove node_modules/ from fec-eregs repo : `rm -rf node_modules/`
9.  Install the requirements: `pip install -r requirements.txt` ` pip install -r requirements_dev.txt`  `pip install -r requirements_test.txt`
10. Install node modules: `npm install` or `npm i`
11. Install all the python pkgs: `npm run build`
12. check if grunt is installed on your machine by running:  ` grunt --version`
13. Install grunt pkg if its not in step#12 by running: `npm install -g grunt-cli`
13. run:  `python manage.py complie_frontend`
14. fec-eregs/local_settings.py : point to the API_BASE = 'https://fec-prod-eregs.app.cloud.gov/regulations/api/ 
15. run: `python manage.py runserver`
16. In the browser go to : http://localhost:8000/ - 45 regs should show on your browser.
